resource-validator/src/main/scala/com/broadinstitute/dsp/resourceValidator/DbReader.scala

```diff
@@ -99,6 +100,40 @@ object DbReader {
           """
       .query[KubernetesCluster]
 
+  // We are calculating the grace period for nodepool deletion assuming that the following are valid proxies for an app's last activity:
+  //    1. destroyedDate for deleted apps
+  //    2. createdDate for error'ed apps
+  // Note that we explicitly check nodepools with 5/11 of statuses that exist.
+  // The statuses we exclude are PROVISIONING, STOPPING, DELETED, PRECREATING, PREDELETING, and PREDELETING
+  //    - We exclude all -ING statuses because they are transitionary, and the purpose of this is not to handle timeouts
+  //    - Unclaimed we exclude because we never want to clean up batch created nodepools
+  // We are excluding default nodepools, as these should remain for the lifetime of the cluster
+  // TODO: Read the grace period (hardcoded to '1 HOUR' below) from config
+  val applessNodepoolQuery =
+    sql"""SELECT np.id, np.nodepoolName, kc.clusterName, kc.googleProject, kc.location
+         FROM NODEPOOL AS np
+         INNER JOIN KUBERNETES_CLUSTER AS kc
+         ON np.clusterId = kc.id
+         WHERE
+            (
+                np.status IN ("STATUS_UNSPECIFIED", "RUNNING", "RECONCILING", "ERROR", "RUNNING_WITH_ERROR")
```

Comment:
Hmmm…I like the explicitness and the ability to copy queries from code and paste them directly onto mysql command lines. At the same time, I'm concerned that if we add/remove a removable status to the ADT, we may not know to update related queries like this. Code robustness/maintenance feels more important to me than development convenience. What do you think?